daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -110,13 +162,47 @@ case class DFDLXXor(recipes: List[CompiledDPath], argType: NodeInfo.Kind) extend
       case NodeInfo.UnsignedShort => arg1.getInt ^ arg2.getInt
       case NodeInfo.UnsignedByte => (arg1.getShort ^ arg2.getShort).toShort
           //$COVERAGE-OFF$
-      case _ => Assert.invariantFailed(s"dfdlx:XOR not supported")
+      case _ => Assert.invariantFailed(s"dfdlx:bitXOR not supported")
           // $COVERAGE-ON$
     }
     }
   }
 
+case class DFDLXOr(recipes: List[CompiledDPath], argType: NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState): DataValuePrimitive = {
+    argType match {
+      case NodeInfo.Long => arg1.getLong | arg2.getLong
+      case NodeInfo.Int => arg1.getInt | arg2.getInt
+      case NodeInfo.Short => (arg1.getShort | arg2.getShort).toShort
+      case NodeInfo.Byte => (arg1.getByte | arg2.getByte).toByte
+      case NodeInfo.UnsignedLong => arg1.getBigInt.or(arg2.getBigInt)
+      case NodeInfo.UnsignedInt => arg1.getLong | arg2.getLong
+      case NodeInfo.UnsignedShort => arg1.getInt | arg2.getInt
+      case NodeInfo.UnsignedByte => (arg1.getShort | arg2.getShort).toShort
+      //$COVERAGE-OFF$
+      case _ => Assert.invariantFailed(s"dfdlx:bitOr not supported")
+      // $COVERAGE-ON$
+    }
+  }
+}
 
+case class DFDLXNot(recipes: CompiledDPath, argType: NodeInfo.Kind) extends FNOneArg(recipes,argType) {
```

Comment:
I think this is a duplicate class? DFDLXBitNot is the right one. I think this can be removed.